Quantity .positive? and negative? failed

Using #positive? and #negative? on quantity, and NumberItem with quantity failed with an error. I would expect that we should be able to use .positive? and negative? on a quantity / numberitem.

Jruby library 3.6.3 on openhab 3.1M4

```
  Scenario Outline: Quantity can use positive? and negative?
    Given code in a rules file
      """
        logger.info("Result is #{<quantity>}")
      """
    When I deploy the rules file
    Then It should log 'Result is <result>' within 5 seconds
    Examples:
      | quantity                          | result |
      | Quantity.new('50 °F').positive?   | true   |
      | Quantity.new('-100 °F').positive? | false  |
      | Quantity.new('50 °F').negative?   | false  |
      | Quantity.new('-100 W').negative?  | true   |
```

Output:
```
08:26:04.232 [INFO ] [jsr223.jruby.Module                  ] - OpenHAB JRuby Scripting Library Version 3.6.3 Loaded
08:26:04.233 [INFO ] [jsr223.jruby.Object                  ] - Processing Started for 5107bace-d831-40ed-906d-83efe6ec4c04
08:26:04.235 [TRACE] [3.jruby.OpenHAB::DSL::Types::Quantity] - Comparing -100 W to 0
08:26:04.235 [TRACE] [3.jruby.OpenHAB::DSL::Types::Quantity] - Coercing -100 W as a request from Integer
08:26:04.236 [TRACE] [3.jruby.OpenHAB::DSL::Types::Quantity] - Unitizing (-100 W) and (0)
08:26:04.250 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab-jruby/tmp/openhab/conf/automation/jsr223/ruby/personal/cucumber_test20210604-21478-q14uhr.rb': org.jruby.embed.EvalFailedException: java.lang.IllegalArgumentException: Can not compare incompatible units.
```

--- a/features/quantity.feature
+++ b/features/quantity.feature
@@ -116,3 +116,32 @@ Feature:  quantity
       | Quantity.new('100 °F') | /        | DecimalType.new(2)   | 50.0 °F  |
       | Quantity.new('50 °F')  | *        | DecimalType.new(2.0) | 100.0 °F |
       | Quantity.new('100 °F') | /        | DecimalType.new(2.0) | 50.0 °F  |
+
+  Scenario Outline: Quantity responds to positive?, negative?, and zero?
+    Given items:
+      | type               | name      | state  | pattern |
+      | Number:Temperature | NumberF   | 2 °F   | %d °F   |
+      | Number:Temperature | NumberC   | 2 °C   | %d °C   |
+      | Number:Power       | PowerPos  | 100 W  |         |
+      | Number:Power       | PowerNeg  | -100 W |         |
+      | Number:Power       | PowerZero | 0 W    |         |
+      | Number             | Number1   | 20     |         |
+    And code in a rules file
+      """
+      logger.info("<object>.<test> #{<object>.<test>}")
+      """
+    When I deploy the rules file
+    Then It should log "<object>.<test> <result>" within 5 seconds
+    Examples:
+      | object                | test      | result |
+      | Quantity.new('50°F')  | positive? | true   |
+      | Quantity.new('-50°F') | negative? | true   |
+      | Quantity.new('10W')   | positive? | true   |
+      | Quantity.new('-1kW')  | positive? | false  |
+      | Quantity.new('0W')    | zero?     | true   |
+      | NumberF               | positive? | true   |
+      | NumberC               | negative? | false  |
+      | PowerPos              | positive? | true   |
+      | PowerNeg              | negative? | true   |
+      | PowerZero             | zero?     | true   |
+      | Number1               | positive? | true   |

--- a/lib/openhab/dsl/types/quantity.rb
+++ b/lib/openhab/dsl/types/quantity.rb
@@ -20,6 +20,7 @@ module OpenHAB
         include OpenHAB::Log
 
         def_delegator :@quantity, :to_s
+        def_delegators '@quantity.double_value', :positive?, :negative?, :zero?
 
         java_import org.openhab.core.library.types.QuantityType
         java_import org.openhab.core.library.types.DecimalType
@@ -185,8 +186,8 @@ module OpenHAB
         #
         # @return [Array]  Array of QuantityType objects
         #
-        def to_qt(*quanities)
-          [quanities].flatten.compact.map { |item| item.is_a?(Quantity) ? item.quantity : item }
+        def to_qt(*quantities)
+          [quantities].flatten.compact.map { |item| item.is_a?(Quantity) ? item.quantity : item }
         end
 
         #

